### ISSUE TITLE
gpg armored ASCII not working on first run

### ISSUE THREAD

**Issue by JohnRDOrazio** [JohnRDOrazio/ansible-wsl-development-setup#4]
We have two code blocks for two different scenarios:

1) a trusted GPG key did not exist initially; in this case we create one and capture the id of the key by running an ad hoc `regex_search` on the `stderr` output of the generation script, then we armor export the key with this id and register the output to `gpg_key_armor`. This correctly gives us a variable `gpg_key_armor` with `stdout` because then we debug it and correctly see the ASCII armor value of this key

2) a trusted GPG key did exist initially; in this case we capture the id of the key by running an ad hoc `regex_search` on the `stdout` output of `gpg --list-secret-keys --keyid-format=long`, then we armor export the key with this id and register the output to `gpg_key_armor`. This also correctly gives us a variable `gpg_key_armor` with `stdout` because then we debug it and correctly see the ASCII armor value of this key

These two cases are mutually exclusive, the code blocks each have an opposite condition: `when: not gpg_key_exists` and `when: gpg_key_exists`.

In the case that a key did not initially exist, the first codeblock is running correctly, because we see all the correct debugging outputs, and the second codeblock is getting skipped correctly, because we see it being skipped.

And yet, for some reason, we then get an error when trying to upload the key to github, saying that `gpg_key_armor` has no `stdout`. Isn't that crazy? It did have `stdout`, we debugged it and output the value just a few steps earlier.

It's almost like the second codeblock that is supposedly skipped, is somehow interfering in some way, and the `gpg_key_armor` value is getting reset.

We'll have to do some further debugging to see what is happening in the meantime...

**Comment by JohnRDOrazio** [comment]
Apparently the second **skipped** code block DOES interfere, here's the debug output after the two mutually exclusive codeblocks:
```console
TASK [Report value of gpg_key_armor once more to make sure it is what we expect it to be] ******************************
ok: [localhost] => {
    "gpg_key_armor": {
        "changed": false,
        "false_condition": "gpg_key_exists",
        "skip_reason": "Conditional result was False",
        "skipped": true
    }
}
```

**Comment by JohnRDOrazio** [comment]
I see that this is an issue that has been raised already in the ansible repo:

https://github.com/ansible/ansible/issues/4297

I'm going to try the workaround [proposed here](https://github.com/ansible/ansible/issues/4297#issuecomment-386962997), rather than trying to register two separate variables and then try to combine them later...

**Comment by JohnRDOrazio** [comment]
Well the workaround proposed did not work, apparently it was behaviour that was permitted for a little while but was then removed again, see here:

https://github.com/ansible/ansible/issues/57839

I guess only way is to use too different variables and then combine the results...

**Comment by JohnRDOrazio** [comment]
fixed in commit 061dd72
